aten/src/ATen/native/MaxPooling.cpp

```diff
@@ -107,8 +98,14 @@ Tensor max_pool1d(
     return std::get<0>(at::max_pool1d_with_indices(
         self, kernel_size, stride, padding, dilation, ceil_mode));
   }
-  return max_pool1d_impl(
-      self, kernel_size, stride, padding, dilation, ceil_mode);
+
+  Tensor result = [&]() {
+    NoNamesGuard guard;
+    return at::_max_pool1d_forward(
```

Comment:
"_max_pool1d_forward_cpu_special" or something to make it clearer that it's a special case